### PR TITLE
Us8 upcoming assignments

### PR DESCRIPTION
**Pull request by vcedgar**
Finished task 141, made student view generic

### CODE REVIEW

**Review by majluni — CHANGES_REQUESTED** [review]
I think it would be good to have some tests for AssignmentDb

File: src/backend/AssignmentDb.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            await this.programDb.update({ role: "Assignment", _id: id }, { name, description, course, assignemnt });
+            return true;
+        }
+        return false;
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    async removeAll() {
+        const programDbPath = Path.relative(process.cwd(), "./data/ProgramDb.db");
+        fs.unlinkSync(programDbPath);
```
Comment (mattfrautnick) [inline src/backend/AssignmentDb.js]: @vcedgar can you look at US10? I just committed how to fix this because as it is now, it does not work.

**Review by mattfrautnick — DISMISSED** [review]
Please update the removeAll for AssignmentsDb, as well as, create tests for AssignmentDb.

**Comment by vcedgar** [comment]
> I think it would be good to have some tests for AssignmentDb

**Comment by zlmonroe** [comment]
I just added the "load assign from db". To test it and see the assignment load, you can add this JSON to the database (the AssignmentDB created this, but I didn't leave any calls since we would just have to delete them later)

```
{"name":"Assignment 2","description":"This is the second assignment. It is a hard assignment worth only 100,000 point.","course":"SER334","assignment":"{}","dueDate":"10/02/13 10:28 AM","role":"Assignment","_id":"5AfDJFNLnJ2HHCjb"}
{"name":"Assignment 1","description":"This is the first assignment. It is a practice assignment worth only 1 point.","course":"SER334","assignment":"{}","dueDate":"10/01/13 10:30 PM","role":"Assignment","_id":"9BdfaQdab7xtvjax"}
```

**Review by majluni — CHANGES_REQUESTED** [review]
I'd say just make it update less often. I'm on mobile so I didn't test it but it looks good

**Review by majluni — APPROVED** [review]

**Pull request merged**